### PR TITLE
Add ActivityStub skeleton

### PR DESCRIPTION
**Pull request by pjanotti**
Bringing the skeleton for the `ActivityStub` described on issue #13, see also https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/18#issuecomment-705157637.

This first PR is just the skeleton for the stub and as such doesn't bring yet dynamic loader and related tests. The API surface is not 100% match with S.D.DS.Activity but gets us ready to start implementing the actual assembly loading. Extending the API to the desired surface is relatively mechanical after the dynamic loading of S.D.DS or the fallback to its "vendored" sources is done.

Some notes:

1. I went through the trouble of making it compatible with the current StyleCop settings for the whole repo, this may not be desirable when we bring the "vendored" sources from dotnet/runtime but for now it keeps the sources consistent.
2. For not the types for `TraceId` and `SpanId` are replaced with `string`: we likely need a wrapper for these types.
3. I didn't bring the methods that match methods from `ActivitySource` my initial thought is that we should also have a separate stub for `ActivitySource` instead of having all of those under `ActivityStub`.
4. For now supplemental data handling was not included either.

Missing `Activity` equivalent API surface:

| Property | S.D.DS.Activity version introduction |
| --------- | ---------------------------------------- |
| ActivityContext | 5.0.0.0 |
| DisplayName | 5.0.0.0 |
| Events | 5.0.0.0 |
| IsAllDataRequested | 5.0.0.0 |
| Links | 5.0.0.0 |
| ParentId | 4.0.2.1 |
| Recorced | 4.0.4.0 |
| Source | 5.0.0.0 |
| StartTimeUtc | 4.0.2.1 |
| TraceStateString | 4.0.4.0 |

| Method | S.D.DS.Activity version introduction |
| --------- | ---------------------------------------- |
| AddEvent | 5.0.0.0 |
| SetEndTime | 4.0.2.1 |
| SetIdFormat | 4.0.4.0 |
| SetParentId (overload w/ 3 parameters) | 4.0.4.0 |
| SetStartTime | 4.0.2.1 |

### CODE REVIEW

File: src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+        /// Provides functionality of <c>System.Diagnostics.DiagnosticSource.Activity.Current</c>.
+        /// Introduction: System.Diagnostics.DiagnosticSource, Version=4.0.2.1
+        /// </remarks>
+        public static ActivityStub Current
+        {
+            get
+            {
+                if (!DynamicLoader.EnsureInitialized())
+                {
+                    // TODO: is the no-op stub enough to represent "no current activity"? no need to differentiate
```
Comment (nrcventura) [inline src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs:44]: Is this `TODO` something we want to start discussing in this PR, or is it a reminder for a future discussion?

File: src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+        {
+            if (ActivityInstance == null)
+            {
+                return;
+            }
+
+            throw new NotImplementedException(MissingCapability);
+        }
+
+        private static class NoOpSingeltons
```
Comment (nrcventura) [inline src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs:570]: Minor typo. It should probably be `NoOpSingletons`.

File: src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs (hunk 1 above)
Comment (pjanotti) [reply src/OpenTelemetry.DynamicActivityBinding/public/ActivityStub.cs:44]: I would like to follow it up as a separate discussion.

**Review by nrcventura — APPROVED** [review]

**Comment by pjanotti** [comment]
Trying to close and re-open to fix the CLA...

**Pull request merged**